packages/flutter_migrate/lib/src/compute.dart

```diff
@@ -0,0 +1,1014 @@
+class MigrateContext {
+  MigrateContext({
+    required this.migrateResult,
+    required this.flutterProject,
+    required this.blacklistPrefixes,
+    required this.logger,
+    required this.verbose,
+    required this.fileSystem,
+    required this.status,
+    required this.migrateUtils,
+    this.baseProject,
+    this.targetProject,
+  });
+
+  MigrateResult migrateResult;
+  FlutterProject flutterProject;
+  Set<String> blacklistPrefixes;
+  Logger logger;
+  bool verbose;
+  FileSystem fileSystem;
+  Status status;
+  MigrateUtils migrateUtils;
+
+  MigrateBaseFlutterProject? baseProject;
+  MigrateTargetFlutterProject? targetProject;
+}
+
+String getLocalPath(String path, String basePath, FileSystem fileSystem) {
+  return path.replaceFirst(basePath + fileSystem.path.separator, '');
+}
+
+String platformToSubdirectoryPrefix(SupportedPlatform platform) {
+  switch (platform) {
+    case SupportedPlatform.android:
+      return 'android';
+    case SupportedPlatform.ios:
+      return 'ios';
+    case SupportedPlatform.linux:
+      return 'linux';
+    case SupportedPlatform.macos:
+      return 'macos';
+    case SupportedPlatform.web:
+      return 'web';
+    case SupportedPlatform.windows:
+      return 'windows';
+    case SupportedPlatform.fuchsia:
+      return 'fuchsia';
+    case SupportedPlatform.root:
+      return 'root';
+  }
+}
+
+/// Computes the changes that migrates the current flutter project to the target revision.
+///
+/// This is the entry point to the core migration computations.
+///
+/// This method attempts to find a base revision, which is the revision of the Flutter SDK
+/// the app was generated with or the last revision the app was migrated to. The base revision
+/// typically comes from the .metadata, but for legacy apps, the config may not exist. In
+/// this case, we fallback to using the revision in .metadata, and if that does not exist, we
+/// use the target revision as the base revision. In the final fallback case, the migration should
+/// still work, but will likely generate slightly less accurate merges.
+///
+/// Operations the computation performs:
+///
+///  - Parse .metadata file
+///  - Collect revisions to use for each platform
+///  - Download each flutter revision and call `flutter create` for each.
+///  - Call `flutter create` with target revision (target is typically current flutter version)
+///  - Diff base revision generated app with target revision generated app
+///  - Compute all newly added files between base and target revisions
+///  - Compute merge of all files that are modified by user and flutter
+///  - Track temp dirs to be deleted
+///
+/// Structure: This method builds upon a MigrateResult instance
+Future<MigrateResult?> computeMigration({
+  bool verbose = false,
+  FlutterProject? flutterProject,
+  String? baseAppPath,
+  String? targetAppPath,
+  String? baseRevision,
+  String? targetRevision,
```

Comment:
It seems like there are some opportunities to extract some objects to make this less of a sea of arguments; e.g., there seem to be two pairs of linked args here that could be extracted to a simple helper.

Reply:
These parameters map one-to-one with user-input command arguments, so it would probably be more clear if they were directly exposed. They aren't the typical internally used variables floating around. There are definitely other cases where it can be cleaned up a bit though.

Reply:
I have pulled out all of the user input parameters into a separate data class.